src/commands/generate/generators/auth/action.js

```diff
@@ -1,9 +1,99 @@
+
 module.exports = async function action (params, utils, command) {
+  let { args } = params
+  let { writeFile, npmCommands } = utils
+  let { installAwsSdk } = npmCommands
+  let error = require('./errors')(params, utils)
   let project = params.inventory.inv._project
   let generate = require('../_generate')
 
-  let manifest = require('./manifest')
 
-  // Run the generic generator
-  generate({ manifest, command, project, utils })
+  let plugins = project.arc.plugins
+  if (plugins.includes('arc-plugin-oauth')) {
+    return error('oauth_plugin_already_installed')
+  }
+
+  let authType = args.t || args.type
+  if (authType === 'oauth') {
+    let manifest = require('./oauth-manifest')
+    generate({ manifest, command, project, utils })
+  }
+  else if (!authType || authType === 'magic-link') {
+
+    const prefsFile = project.localPreferencesFile
+    const { readFileSync } = require('fs')
+    let prefs = readFileSync(prefsFile, 'utf8')
+    if (!/@sandbox-startup/.test(prefs)) {
+      prefs += `@sandbox-startup
+node ./scripts/seed-users.js`
+      writeFile(prefsFile, prefs)
+    }
```

Comment:
So, I think you can change this up a bit and check to see if `params.inventory._project.localPreferences` has the key `sandbox-startup`. If it doesn't, you can just append it to the prefs file like you are doing, but if it does exist, you can check to see if `node ./scripts/seed-users.js` is already in the array. If not, you will probably need to figure out a way to append to the `sandbox-startup` section.

Reply:
Done.